File: microservices/investment/src/main/java/com/illunex/invest/investment/persistence/entity/ReviewItemCategory.java
package com.illunex.invest.investment.persistence.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "review_item_category")
public class ReviewItemCategory {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idx;
    String category;

    @ManyToOne @JoinColumn(name = "template_idx")
    ReviewItemTemplate reviewItemTemplate;

    @OneToMany(mappedBy = "reviewItemCategory", cascade = CascadeType.ALL)
    List<ReviewItem> reviewItem;
}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/ReviewItemCategory.kt
package com.illunex.invest.invest.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "review_category")
data class ReviewItemCategory(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null,
    var vcCompanyIdx: Long? = null,
    var category: String? = null,

    @OneToMany(mappedBy = "category")
    var item: List<ReviewItem>?

) {
    constructor() : this(null, null, null, null)
}
